aquacomputer_d5next.c

```diff
@@ -654,6 +691,413 @@ static void aqc_debugfs_init(struct aqc_data *priv)
 
 #endif
 
+
+static int aqc_fan_curve_show(struct device *dev, struct device_attribute *attr, int offset)
+{
+	struct sensor_device_attribute_2 *sensor_attr = to_sensor_dev_attr_2(attr);
+	struct aqc_data *priv = dev_get_drvdata(dev);
+	int channel = sensor_attr->nr;
+	int idx = sensor_attr->index;
+	u16 to_get;
+
+	if (channel > priv->num_fans)
+		return -ENOENT;
+	if ((idx > NUM_CTRL_CURVE_POINTS) || idx < 0)
+		return -ENOENT;
+
+	to_get = priv->fan_ctrl_offsets[channel] + offset + 2*idx;
+
+	return aqc_get_ctrl_val(priv, to_get);
+}
+
+static ssize_t aqc_fan_curve_pwm_show(struct device *dev, struct device_attribute *attr, char *buf)
+{
+	int val = aqc_fan_curve_show(dev, attr, AQC_FAN_CTRL_CURVE_PWMS_OFFSET);
+	if (val < 0)
+		return val;
+	return sprintf(buf, "%d\n", aqc_percent_to_pwm(val));
+}
+
+static ssize_t aqc_fan_curve_temp_show(struct device *dev, struct device_attribute *attr, char *buf)
+{
+	int val = aqc_fan_curve_show(dev, attr, AQC_FAN_CTRL_CURVE_TEMPS_OFFSET);
+	if (val < 0)
+		return val;
+	return sprintf(buf, "%d0\n", val);
+}
+
+static int aqc_fan_curve_store(struct device *dev, struct device_attribute *attr, int offset, long val)
+{
+	struct sensor_device_attribute_2 *sensor_attr = to_sensor_dev_attr_2(attr);
+	struct aqc_data *priv = dev_get_drvdata(dev);
+	int channel = sensor_attr->nr;
+	int idx = sensor_attr->index;
+	u16 to_set;
+
+	if (channel > priv->num_fans)
+		return -ENOENT;
+	if ((idx > NUM_CTRL_CURVE_POINTS) || idx < 0)
+		return -ENOENT;
+
+	to_set = priv->fan_ctrl_offsets[channel] + offset + 2*idx;
+
+	return aqc_set_ctrl_val(priv, to_set, val);
+}
+
+static ssize_t aqc_fan_curve_pwm_store(struct device *dev, struct device_attribute *attr, const char *buf, size_t count)
+{
+	long val;
+	if (kstrtol(buf, 10, &val) < 0)
+		return -EINVAL;
+	if (val < 0 || val > 255)
+		return -EINVAL;
+
+	val = aqc_pwm_to_percent(val);
+
+	return aqc_fan_curve_store(dev, attr, AQC_FAN_CTRL_CURVE_PWMS_OFFSET, val);
+}
+
+static ssize_t aqc_fan_curve_temp_store(struct device *dev, struct device_attribute *attr, const char *buf, size_t count)
+{
+	long val;
+	if (kstrtol(buf, 10, &val) < 0)
+		return -EINVAL;
+
+	val = val / 10;
+
+	return aqc_fan_curve_store(dev, attr, AQC_FAN_CTRL_CURVE_TEMPS_OFFSET, val);
+}
+
+
+static SENSOR_DEVICE_ATTR_2_RW(pwm0_auto_point1_pwm, aqc_fan_curve_pwm, 0, 1);
```

Comment:
Sysfs starts pwm1 so we should do that here. 
idx should start from 0 to simplify offset calculation.
This line would be:
static SENSOR_DEVICE_ATTR_2_RW(pwm1_auto_point1_pwm, aqc_fan_curve_pwm, 0, 0);

Reply:
also fixed in f65a7cd3